### DISCUSSION TITLE
How to fetch specific folder from repo.

### DISCUSSION THREAD

**Discussion by dir1337** [community/community#75838] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hello,
Respected devs,

**Problem:**
How it is possible to fetch a particular/desired folder from the repo.

**Example**
In order to fetch following folder what steps i need to follow.  

https://github.com/mdn/learning-area/tree/main/javascript/oojs/bouncing-balls
[ github -> mdn(\<org>) -> learning-area.git(\<repo>) -> javascript -> oojs -> bouncing-balls(\<desired_folder>)]

**Comment by sky9262** [comment, marked as answer]
Fetching a specific directory from a GitHub repository is not directly supported by Git, but you can achieve this using a combination of Git and Subversion (SVN) commands. Here’s how you can do it:

## Install SVN
For Windows: https://tortoisesvn.net/downloads.html

for Linux:
```bash
sudo apt install subversion
```
OR
Follow this link https://subversion.apache.org/


Navigate to the folder you want to download in your browser. In your case, it’s the `bouncing-balls` folder in the `learning-area` repository.

Modify the URL for SVN. Replace `tree/main` with `trunk`. So, your URL will look like this: `https://github.com/mdn/learning-area/trunk/javascript/oojs/bouncing-balls`.

Open your terminal or command line interface, and use the SVN command to checkout the folder:

```bash
svn checkout https://github.com/mdn/learning-area/trunk/javascript/oojs/bouncing-balls
```

*This will download the bouncing-balls folder to your current directory*

> Please note that this method downloads the folder as a standalone directory, and it won’t be linked to the original GitHub repository. If you want to make changes and push them back to the repository, you’ll need to clone the entire repository, make your changes, and then push the changes back


## Result: 
![image](https://github.com/community/community/assets/68050118/7503a032-b156-441f-8bb6-08d225918b6a)

**Reply by dir1337** [reply]
Thank you @sky9262, For the hint of "subversion".
Your suggested tortoise svn is limited for windows system.
I suggest you to please add the https://subversion.apache.org/  in your post(by editing). So I can mark it as "answer".

**Comment by leith** [comment]
With SVN support removed just 2 months after this answer ([see the GitHub blog](https://github.blog/news-insights/product-news/sunsetting-subversion-support/)), there is currently no official way of doing this. The [GitHub API](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#get-repository-content) will list `null` as a download link for any directory when getting repository content, and there is no `folder` or `path` parameter when using the [download a repository archive](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#download-a-repository-archive-tar) endpoint.

There are a few third party solutions that have been developed ([see this StackOverflow](https://stackoverflow.com/q/7106012/2053165)), but they all have various limitations.

Using git's sparse checkout feature is useful if you simply want to clone something, however there are times when an archive of a sub-folder is useful to retrieve directly - particularly for monorepos containing multiple packages and automation workflows. [Composer faces this challenge](https://github.com/composer/packagist/issues/472) when trying to support monorepo development for the open source community and developers end up with a lot of very strange workarounds to propagate into read-only repos to get around it.
